### ISSUE TITLE
onRequest: calling deny() halts all DOM rendering and JavaScript execution

### ISSUE THREAD

**Issue by dansilivestru** [blackberry/Ripple-Framework#2]
Having the following in the container window:

```
window.onRequest = function (request) {
    if (request.url === "http://www.google.com") {
        request.deny();
    else {
        request.allow();
     }
};
```

And adding an image in the emulation window like this:

```
<img src="http://foo.com/img.jpg" />
```

When executing request.deny() all DOM parsing and JavaScript execution fails.

**Comment by ghost** [comment]
if there are alert() s, request.deny() crashes. Working on this now

**Comment by ghost** [comment]
alert() call in onReqeust's callback changed the order of serverPendingRequest() and requestTimerFired().
confirm() and prompt() may have the same problem.
Fix will be in the new build of qtwebkit.

**Comment by dansilivestru** [comment]
Do you want to create an issue for confirm() and prompt() as well as details on where the problematic code might be. Once you've tested them and confirmed that they suffer from the same problem.

**Comment by ghost** [comment]
yes, I just tested prompt() and confirm() calls. They crash the app too.
